agent/service.go

```diff
@@ -154,7 +155,25 @@ func (ts *TaskService) Create(requestCtx context.Context, req *taskAPI.CreateTas
 		return nil, fmt.Errorf("Drive %q could not be found", driveID)
 	}
 
-	err = bundleDir.MountRootfs(drive.Path(), "ext4", nil)
+	const (
+		maxRetries = 3
+		retryDelay = 25 * time.Millisecond
+	)
+
+	// We retry here due to Firecracker needing some time to populate the guest
+	// drive.
+	// https://github.com/firecracker-microvm/firecracker/issues/1159
+	for i := 0; i < maxRetries; i++ {
+		if err = bundleDir.MountRootfs(drive.Path(), "ext4", nil); err != nil {
```

Comment:
Can we only do a retry if err is `EINVAL` so we don't retry when some unrelated error occurs?

Reply:
Good call! Fixed